src/features/upload-file/upload-file.feature.ts

```diff
@@ -16,15 +16,28 @@ import UploadOptions, { UploadOptionsWithDefault } from './types/upload-options.
 import { fillRecordWithPath } from './utils/fill-record-with-path'
 import { getProvider } from './utils/get-provider'
 
-export type ProviderOptions = Required<
-  Exclude<UploadOptions['provider'], BaseProvider>
->;
-
 const DEFAULT_FILE_PROPERTY = 'file'
 const DEFAULT_FILE_PATH_PROPERTY = 'filePath'
 const DEFAULT_FILES_TO_DELETE_PROPERTY = 'filesToDelete'
 
+export function registerComponents(componentLoader?: ComponentLoader) {
```

Comment:
this function gets exported so the components can be registered before the feature is added to a resource